### ISSUE TITLE
TinfourDemo finds no ground points even they are there

### ISSUE THREAD

**Issue by zdila** [gwlucastrig/Tinfour#90]
When I open las file with classified ground points with TinfourDemo, it outputs an error:

```
Error loading tile_-235000_-1227000.las Source Lidar file does not contain samples for GroundPoints
```

I see that the ground points are present with QGIS: 

![image](https://user-images.githubusercontent.com/636095/195410260-cba1040f-97bb-4587-809c-30e2481a0585.png)

**Comment by gwlucastrig** [comment]
I'm going to take a look at this tomorrow.  Is there any chance you can post the LAS file for me to look at?  (The LAZ file from your previous issue was a big help).

**Comment by zdila** [comment]
Sample file (please ungzip it before opening ;-)) https://drive.google.com/file/d/14FhA1AZeL0cqFZLTzUaTVvV_2-_IuB_-/view?usp=sharing

Note that the license of the file (and other I provided in previous issue) is mentioned at [https://www.geoportal.sk/files/zbgis/objednavky/podmienky-poskytovania-pouzivania-produktov-lls.pdf (Slovak language only)](https://www.geoportal.sk/files/zbgis/objednavky/terms_and_conditions_of_provision_and_use_als.pdf)

**Comment by gwlucastrig** [comment]
Thanks for the sample data.  The problem with the point classification was due to the fact that your file was in LAS version 1.4 format and used record type 6, which Tinfour did not support.  When I wrote the original Tinfour LAS file reader classes, I was unable to find any sample files in version 1.4 format.  So I was unable to test it using version 1.4 files.

Anyway, I am grateful to you for finding this problem and posting your sample data.

I have more testing to do. When it is finished, I upload the new code.

![Issue90](https://user-images.githubusercontent.com/19698145/195734896-00e607f7-bf27-4de3-9acc-7b6b5816e5a5.png)
